File: network-configuration-server/src/main/java/en/ubb/networkconfiguration/boundary/dto/network/virtual/VirtualLayerDto.java
package en.ubb.networkconfiguration.boundary.dto.network.virtual;

import com.fasterxml.jackson.annotation.JsonProperty;
import en.ubb.networkconfiguration.persistence.domain.network.enums.LayerType;
import lombok.*;

import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Setter
public class VirtualLayerDto {

    @JsonProperty("id")
    private Long id;

    @JsonProperty("type")
    private LayerType type;

    @JsonProperty("nodes")
    private List<VirtualNodeDto> nodes;
}

File: client-app/src/app/shared/models/network/virtual/VirtualLayerDto.ts
import {VirtualNodeDto} from './VirtualNodeDto';
import {LayerType} from '../shared/LayerType';
import {Activation} from '../shared/Activation';

export interface VirtualLayerDto {
  id: number;
  type: LayerType;
  activation: Activation;
  nodes: VirtualNodeDto[];
}
